indico/modules/events/reminders/models/reminders.py

```diff
@@ -26,10 +28,18 @@
 from indico.modules.events.reminders.util import get_reminder_email_tpl
 from indico.util.date_time import now_utc
 from indico.util.enum import IndicoIntEnum
+from indico.util.placeholders import replace_placeholders
 from indico.util.signals import values_from_signal
 from indico.util.string import format_repr
 
 
+@dataclass(frozen=True)
+class Recipient:
+    email: str
+    first_name: str = field(default='')
+    last_name: str = field(default='')
```

Comment:
IIRC you do not need `field` for simple defaults (but only for more complex cases such as mutable objects where you need a factory for the default):

```suggestion
    first_name: str = ''
    last_name: str = ''
```